android/app/src/bt/java/org/pathcheck/covidsafepaths/MainActivity.java

```diff
@@ -139,4 +172,97 @@ private void handleExposureStateChanged(boolean enabled) {
 
     }
   }
-}
\ No newline at end of file
+
+  public void share() {
+    FluentFuture<Boolean> getKeysAndSubmitToService =
+            FluentFuture.from(getRecentKeys())
+                    .transform(
+                            this::toDiagnosisKeysWithTransmissionRisk, AppExecutors.getLightweightExecutor())
+                    .transformAsync(this::submitKeysToService, AppExecutors.getBackgroundExecutor());
+
+    Futures.addCallback(
+            getKeysAndSubmitToService,
+            new FutureCallback<Boolean>() {
+              @Override
+              public void onSuccess(Boolean shared) {
+//                        sharedLiveEvent.postValue(shared);
+//                        postInflight(false);
+              }
+
+              @Override
+              public void onFailure(Throwable exception) {
+                if (!(exception instanceof ApiException)) {
+//                            Log.e(TAG, "Unknown error", exception);
+//                            snackbarLiveEvent.postValue(
+//                                    getApplication().getString(R.string.generic_error_message));
+//                            postInflight(false);
+                  return;
+                }
+                ApiException apiException = (ApiException) exception;
+                if (apiException.getStatusCode()
+                        == ExposureNotificationStatusCodes.RESOLUTION_REQUIRED) {
+                  showPermissionShareKeys(apiException);
+                } else {
+//                            Log.w(TAG, "No RESOLUTION_REQUIRED in result", apiException);
```

Comment:
Is this code meant to be commented out? Could we just remove it instead and perhaps rid of the else branch?